projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-react-page.php

```diff
@@ -181,6 +181,41 @@ function page_admin_scripts() {
 		wp_add_inline_script( 'react-plugin', 'var jetpack_redirects = { currentSiteRawUrl: "' . $site_suffix . '" };', 'before' );
 	}
 
+	/**
+	 * Gets a purchase token that is used for Jetpack logged out visitor checkout.
+	 * The purchase token should be appended to all CTA url's that lead to checkout.
+	 *
+	 * @since 9.6.3
+	 * @return string|boolean
+	 */
+	public function getPurchaseToken() {
+		if ( ! Jetpack::current_user_can_purchase() ) {
+			return false;
+		}
+
+		$purchase_token = Jetpack_Options::get_option( 'purchase_token', false );
+
+		if ( $purchase_token ) {
+			return $purchase_token;
+		}
+		// If the purchase token is not saved in the options table yet, then add it.
+		Jetpack_Options::update_option( 'purchase_token', $this->generatePurchaseToken(), true );
+		return Jetpack_Options::get_option( 'purchase_token', false );
+	}
+
+	/**
+	 * Generates a purchase token that is used for Jetpack logged out visitor checkout.
+	 *
+	 * @since 9.6.3
+	 * @return string
+	 */
+	public function generatePurchaseToken() {
+		if ( function_exists( 'wp_generate_password' ) ) {
```

Comment:
Do we need the `function_exists` check here? It seems `wp_generate_password` was added to WP quite a while ago now.